src/components/Trends/TrendsTable/TrendsTable.js

```diff
@@ -37,9 +44,17 @@ const NumberCircle = props => (
 
 class TrendsTable extends PureComponent {
   state = {
+    selected: this.props.selectedTrends,
     connectedTrends: []
   }
 
+  componentWillUnmount () {
+    const { selected } = this.state
+    if (selected.size > 0) {
+      this.props.dispatchSelectedTrends(selected)
```

Comment:
use 'dispatch' in a name for dispatch function is not very good. Or Did I miss any trend? @DmitriVanGuard

Reply:
Was not able to come up with a better name